src/serverjar/serverjar_paper.py

```diff
@@ -104,11 +111,48 @@ def getDownloadFileName(paperMcVersion, paperBuild):
 
 def paperCheckForUpdate(installedServerjarFullName):
     mcVersion = getInstalledPaperMinecraftVersion(installedServerjarFullName)
+
+    # Report an error if getInstalledPaperMinecraftVersion encountered an issue.
+    if not mcVersion:
+        print(oColors.brightRed + f"ERR: An error was encountered while detecting the server's Minecraft version." +
+              oColors.standardWhite)
+        return False
+
     paperInstalledBuild = getInstalledPaperVersion(installedServerjarFullName)
+
+    # Report an error if getInstalledPaperVersion encountered an issue.
+    if not paperInstalledBuild:
+        print(oColors.brightRed + f"ERR: An error was encountered while detecting the server's Paper version." +
+              oColors.standardWhite)
+        return False
+
     versionGroup = findVersionGroup(mcVersion)
+
+    # Report an error if findVersionGroup encountered an issue.
+    if not versionGroup:
+        print(oColors.brightRed + f"ERR: An error was encountered while fetching the server's version group." +
+              oColors.standardWhite)
+        return False
+
     paperLatestBuild = findLatestBuild(versionGroup)
+
+    # Report an error if findLatestBuild encountered an issue.
+    if not paperLatestBuild:
+        print(oColors.brightRed + f"ERR: An error was encountered while fetching the latest version of PaperMC." +
+              oColors.standardWhite)
+        return False  # Not currently handled, but can be at a later date. Currently just stops the following from
+    #                   being printed.
+
     paperVersionBehind = versionBehind(paperInstalledBuild, paperLatestBuild)
 
+    # Report an error if getInstalledPaperVersion encountered an issue.
+    if not paperVersionBehind:
+        print(oColors.brightRed + f"ERR: An error was encountered while detecting how many versions behind you are. "
```

Comment:
False is a little bit unclear when not detecting how many versions you are behind. 
Use 'N/A' to make it clearer and unified with the rest of the program.

```suggestion
paperVersionBehind = 'N/A'
print(oColors.brightRed + f"ERR: An error was encountered while detecting how many versions you are behind. "
                          f"Will display as N/A." + oColors.standardWhite)
```